pygitguardian/models.py

```diff
@@ -485,6 +485,99 @@ def __repr__(self) -> str:
         return f"content:{repr(self.content)}"
 
 
+class HoneytokenResponseSchema(BaseSchema):
+    id = fields.Int()
+    name = fields.String()
+    description = fields.String(allow_none=True)
+    created_at = fields.AwareDateTime()
+    status = fields.String()
+    triggered_at = fields.AwareDateTime(allow_none=True)
+    revoked_at = fields.AwareDateTime(allow_none=True)
+    open_events_count = fields.Int(allow_none=True)
+    type_ = fields.String(data_key="type")
+    creator_id = fields.Int(allow_none=True)
+    revoker_id = fields.Int(allow_none=True)
+    creator_api_token_id = fields.String(allow_none=True)
+    revoker_api_token_id = fields.String(allow_none=True)
+    token = fields.Mapping(key=fields.String(), value=fields.String())
+    tags = fields.List(fields.String())
+
+    @post_load
+    def make_honeytoken_response(
+        self, data: Dict[str, Any], **kwargs: Any
+    ) -> "HoneytokenResponse":
+        return HoneytokenResponse(**data)
+
+
+class HoneytokenResponse(Base):
+    """
+    honeytoken creation in the GitGuardian API.
+    Allows users to create and get a honeytoken.
+    Example:
+        {
+            "id": 141,
+            "name": "honeytoken A",
+            "description": "honeytoken used in the repository AA",
+            "created_at": "2019-08-22T14:15:22Z",
+            "status": "active",
+            "triggered_at": "2019-08-22T14:15:22Z",
+            "revoked_at": "2019-08-22T14:15:22Z",
+            "open_events_count": 122,
+            "type": "AWS",
+            "creator_id": 122,
+            "revoker_id": 122,
+            "creator_api_token_id": null,
+            "revoker_api_token_id": null,
+            "token": {
+                "access_token_id": "AAAA",
+                "secret_key": "BBB"
+            },
+        "tags": ["publicly_exposed"]
+        }
+    """
+
+    SCHEMA = HoneytokenResponseSchema()
+
+    def __init__(
+        self,
+        id,
+        name,
+        description,
+        created_at,
+        status,
+        triggered_at,
+        revoked_at,
+        open_events_count,
+        type_,
+        creator_id,
+        revoker_id,
+        creator_api_token_id,
+        revoker_api_token_id,
+        token,
+        tags,
```

Comment:
These are missing type-hinting. Did you try the marshmallow_dataclass approach? It can save you some typing :)

Reply:
no sure to make optional or and timezone aware datetime. And I don't want to spend ages in doc as this already works. I will just add the typing here.

Reply:
makes sense.